src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringConcat.cs

```diff
@@ -397,6 +706,21 @@ private BoundExpression ConvertConcatExprToString(SyntaxNode syntax, BoundExpres
                 }
             }
 
+            // If expression is of form `constantChar.ToString()` then rewrite it here to a string literal so we can choose better options for lowering later (e.g. fold it with another constant instead of performing concatenation)
+            // NOTE: We request `object.ToString()` as an optional member here to avoid reporting "missing member" diagnostic earlier than we should.
+            // Because if in the end it turns out that we have a simple concatenation of n strings there is no need to report missing `object.ToString()` member
+            if (TryGetSpecialTypeMethod(expr.Syntax, SpecialMember.System_Object__ToString, out MethodSymbol? optionalObjectToString, isOptional: true) &&
+                expr is BoundCall { ReceiverOpt: { Type: NamedTypeSymbol { SpecialType: SpecialType.System_Char } charType, ConstantValueOpt: { IsChar: true } charConstant } } call &&
+                call.Method.GetLeastOverriddenMember(charType) == optionalObjectToString)
+            {
+                var oldSyntax = _factory.Syntax;
+                _factory.Syntax = expr.Syntax;
+                expr = _factory.Literal(charConstant.CharValue.ToString());
```

Comment:
>expr = _factory.Literal(charConstant.CharValue.ToString()); [](http://example.com/codeflow?start=16&length=59)

I suggest to avoid doing this work in two paces and instead placing this `if` before the previous one. And simply unwrap that character constant, letting the `if` above to handle the constant conversion.